### DISCUSSION TITLE
Custom credential session properties

### DISCUSSION THREAD

**Discussion by ta-vroom** [nextauthjs/next-auth#4532] [Help]
I have read [this section](https://next-auth.js.org/getting-started/typescript#module-augmentation) many times over, but I cannot get it to work.

I am trying to use these augmentations: 

```
///types/next-auth.d.ts
import NextAuth from "next-auth"

declare module "next-auth" {
  interface Session {
    user: {
      name: string,
      address: string
    }
  }
}
```

And here is my authorize function: 
```
///[...nextAuth].ts
...
authorize: async (credentials, req) => {
    const res = await fetch('http://localhost:2000/login', {
        method: 'POST',
        body: JSON.stringify(credentials),
        headers: { 'Content-Type': 'application/json' },
    });
    const user = await res.json();
    if (res.ok && user) {
      return {name: "JSomething", address: "42 Wallaby Way"};
}
```

I **expect** that when I go to "/api/auth/session" the user object would contain { name: "JSomething", address "42 Wallaby Way" }, but instead I only get {name: "JSomething" }. I also added typeRoot to my tsconfig.json and am including the types path in include.

**Comment by benlongo** [comment]
Also having problems with this. I cannot get the types to work.
